src/Core/System/SalesChannel/SalesChannel/ContextSwitchRoute.php

```diff
@@ -76,6 +77,10 @@ public function switchContext(RequestDataBag $data, SalesChannelContext $context
             ->add(self::STATE_ID, new Type('string'))
         ;
 
+        $event = new SwitchContextEvent($data, $definition, $parameters, $context);
```

Comment:
would be nice, if you could use our new way of dispatching events https://github.com/shopware/shopware/blob/trunk/adr/2024-06-18-extended-event-system.md

you might probably hear of this :wink:

Reply:
IMO this is not correct. The new system is designed for pre / post / replace functions. But here you just want a "notify"